src/libraries/System.Private.CoreLib/src/System/Runtime/InteropServices/Architecture.cs

```diff
@@ -3,16 +3,49 @@
 
 namespace System.Runtime.InteropServices
 {
+    /// <summary>
+    /// Indicates the processor architecture.
+    /// </summary>
     public enum Architecture
     {
+        /// <summary>
+        /// An Intel-based 32-bit processor architecture.
+        /// </summary>
         X86,
+        /// <summary>
+        /// An Intel-based 64-bit processor architecture.
+        /// </summary>
         X64,
+        /// <summary>
+        /// A 32-bit ARM processor architecture.
+        /// </summary>
```

Comment:
If we're including the specifics in the remarks of `Armv6`, we should include at least something here to specify how it differs. I've explicitly excluded mentioning 'hard-float ABI' support as the 'armel' architecture that is community supported and used by Samsung (that uses the soft-float ABI) also uses the `Arm` value here to my knowledge.

```suggestion
        /// A 32-bit ARM processor architecture.
        /// </summary>
        /// <remarks>
        /// This value indicates ARMv7 base instructions, VFPv3 floating point support and registers, and Thumb2 compact instruction set.
        /// </remarks>
```

Reply:
Are we sure that this is true? `Arm` has been there a while, has it always meant the above? I think we might want to make this the latest `Arm` support and add new ones if we have narrower constraints.

Reply:
> If we're including the specifics in the remarks of Armv6

That doesn't follow. `Arm` is intentionally generic whereas `Armv6` was introduced specifically to provide clarity. If we go with the above suggestion `Arm` now means `Armv7`, which would be bad if we decided to support something new and `Arm` means a specific thing as opposed to simply "latest". That could be the intent, but I'd like to hear from the ARM scenario stack holders about what makes sense for them.

Reply:
I just checked the .NET Core 2.1.0 release ([the first version to support ARM32](https://github.com/dotnet/runtime/issues/8738)) tag in dotnet/coreclr, and the build points to ARMv7 instructions with VFPv3 and Thumb2: https://github.com/dotnet/coreclr/blob/v2.1.0/cross/toolchain.cmake#L84-L102

Given [the tentative plan to drop ARM32 support in .NET 9](https://github.com/dotnet/runtime/discussions/71042), I don't think we're going to be redefining the minimum supported version for the ARM32 instruction set to something newer any time soon.

Reply:
Let's hear from the stake holders. If they are good with it so am I.

Reply:
I've been operating on the model that `Arm` means `Armv7`/`Arm32`. I think that's fine. There isn't a roll-forward concept for code, whether `Arm` could eventually mean `Armv8`+/`Arm64`.